examples/self-supervised-learning/demo_ei_transforms.py

```diff
@@ -0,0 +1,158 @@
+-  Rotate: 2D image rotation;
+-  Scale: continuous 2D image downscaling;
+-  Euclidean: includes continuous translation, rotation, and reflection,
+   forming the group :math:`\mathbb{E}(2)`;
+-  Similarity: as above but includes scale, forming the group
+   :math:`\text{S}(2)`;
+-  Affine: as above but includes shear effects, forming the group
+   :math:`\text{Aff}(3)`;
+-  Homography: as above but includes perspective (i.e pan and tilt)
+   effects, forming the group :math:`\text{PGL}(3)`;
+-  PanTiltRotate: pure 3D camera rotation i.e pan, tilt and 2D image
+   rotation.
+
+These were proposed in the papers:
+
+-  ``Shift``, ``Rotate``: `Chen et al., Equivariant Imaging: Learning
+   Beyond the Range
+   Space <https://openaccess.thecvf.com/content/ICCV2021/papers/Chen_Equivariant_Imaging_Learning_Beyond_the_Range_Space_ICCV_2021_paper.pdf>`__
+-  ``Scale``: `Scanvic et al., Self-Supervised Learning for Image
+   Super-Resolution and Deblurring <https://arxiv.org/abs/2312.11232>`__
+-  ``Homography`` and the projective geometry framework: `Wang et al.,
+   Perspective-Equivariant Imaging: an Unsupervised Framework for
+   Multispectral Pansharpening <https://arxiv.org/abs/2403.09327>`__
+
+"""
+
+import deepinv as dinv
+import torch
+from torch.utils.data import DataLoader, random_split
+from torchvision.datasets import ImageFolder
+from torchvision.transforms import Compose, ToTensor, CenterCrop, Resize
+from torchvision.datasets.utils import download_and_extract_archive
+
+device = dinv.utils.get_freer_gpu() if torch.cuda.is_available() else "cpu"
+
+
+# %%
+# Define transforms. For the transforms that involve 3D camera rotation
+# (i.e pan or tilt), we limit ``theta_max`` for display.
+#
+
+transforms = [
+    dinv.transform.Shift(),
+    dinv.transform.Rotate(),
+    dinv.transform.Scale(),
+    dinv.transform.Homography(theta_max=10),
+    dinv.transform.projective.Euclidean(),
+    dinv.transform.projective.Similarity(),
+    dinv.transform.projective.Affine(),
+    dinv.transform.projective.PanTiltRotate(theta_max=10),
+]
+
+
+# %%
+# Plot transforms on a sample image. Note that, during training, we never
+# have access to these ground truth images ``x``, only partial and noisy
+# measurements ``y``.
+#
+
+x = dinv.utils.load_url_image(dinv.utils.demo.get_image_url("celeba_example.jpg"))
+dinv.utils.plot(
+    [x] + [t(x) for t in transforms],
+    ["Orig"] + [t.__class__.__name__ for t in transforms],
+)
+
+
+# %%
+# Now, we run an inpainting experiment to reconstruct images from images
+# masked with a random mask, without ground truth, using EI. For this
+# example we use the Urban100 images of natural urban scenes. As these
+# scenes are imaged with a camera free to move and rotate in the world,
+# all of the above transformations are valid invariances that we can
+# impose on the unknown image set :math:`x\in X`.
+#
+
+physics = dinv.physics.Inpainting((3, 256, 256), mask=0.6, device=device)
+
+download_and_extract_archive(
```

Comment:
can you please update to the Urban100HR class added in a recent PR?

Reply:
This increases demo runtime by ~1min as the current deepinv Urban100HR is much bigger than the HF Urban100 I'm using. I could create an issue?